### ISSUE TITLE
Some questions about model KTMR

### ISSUE THREAD

**Issue by rui-919** [BITwzl/KTMR#1]
Hi, the class KTMR in the file model_swinir.py, I'm not sure how DCL is implemented. As well as in the file model_ktmr.py "out_dict['L'] =consistency_layer(self.L.detach()[0].float().cpu(), self.H.detach()[0].float().cpu(), self.L.mask.squeeze()).reshape(self.L.detach()). self.mask.squeeze()).reshape(self.L.detach()[0].shape)" How to interpret.

**Comment by BITwzl** [comment]
Glad to answer your question!
The net_swinir.py is the network structure without the DCL. The DCL is implemented in utils_rec.py and used in current_visuals()@model_ktmr.py. This function will be invoked after the network forwarding, before the computation of loss function in training, and the writing of inference results.
